src/Core/Geometry/IndexedGeometry.cpp

```diff
@@ -0,0 +1,254 @@
+#include <Core/Geometry/IndexedGeometry.hpp>
+#include <iterator>
+
+namespace Ra {
+namespace Core {
+namespace Geometry {
+
+/// \fixme Deep copy
+MultiIndexedGeometry::MultiIndexedGeometry( const MultiIndexedGeometry& other ) :
+    AttribArrayGeometry( other ), m_indices( other.m_indices ) {}
+
+MultiIndexedGeometry::MultiIndexedGeometry( MultiIndexedGeometry&& other ) :
+    AttribArrayGeometry( std::move( other ) ), m_indices( std::move( other.m_indices ) ) {}
+
+MultiIndexedGeometry::MultiIndexedGeometry( const AttribArrayGeometry& other ) :
+    AttribArrayGeometry( other ) {
+#ifdef MULTI_INDEX_MIMIC_TRIANGLE_MESH
+    setIndices( {} ); // force create of trianglemesh layer
+#endif
+}
+
+MultiIndexedGeometry::MultiIndexedGeometry( AttribArrayGeometry&& other ) :
+    AttribArrayGeometry( std::move( other ) ) {
+#ifdef MULTI_INDEX_MIMIC_TRIANGLE_MESH
+    setIndices( {} ); // force create of trianglemesh layer
+#endif
+}
+
+/// \fixme Deep copy
+MultiIndexedGeometry& MultiIndexedGeometry::operator=( const MultiIndexedGeometry& other ) {
+    AttribArrayGeometry::operator=( other );
+    m_indices                    = other.m_indices;
+    notify();
+    return *this;
+}
+
+/// \fixme Deep copy
+MultiIndexedGeometry& MultiIndexedGeometry::operator=( MultiIndexedGeometry&& other ) {
+    AttribArrayGeometry::operator=( std::move( other ) );
+    m_indices                    = std::move( other.m_indices );
+    notify();
+    return *this;
+}
+
+void MultiIndexedGeometry::clear() {
+    m_indices.clear();
+    AttribArrayGeometry::clear();
+    notify();
+}
+
+/// \fixme Deep copy
+void MultiIndexedGeometry::copy( const MultiIndexedGeometry& other ) {
+    AttribArrayGeometry::copyBaseGeometry( other );
+    m_indices = other.m_indices;
+    notify();
+}
+
+/// \todo Implement MultiIndexedGeometry::checkConsistency
+void MultiIndexedGeometry::checkConsistency() const {
+#ifdef CORE_DEBUG
+#endif
+}
+
+bool
+/// \todo Implement MultiIndexedGeometry::append
+MultiIndexedGeometry::append( const MultiIndexedGeometry& other ) {
+    bool isCopyComplete = true;
+    for ( const auto& [key, value] : m_indices )
+    {
+        auto it = other.m_indices.find( key );
+        if ( !( it == other.m_indices.end() && value.second->append( *( it->second.second ) ) ) )
+            isCopyComplete = false;
+    }
+
+    return isCopyComplete;
+}
+
+//////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////
+
+bool MultiIndexedGeometry::containsLayer( const LayerSemantic& semanticName ) const {
+    for ( const auto& [key, value] : m_indices )
+    {
+        if ( key.first.find( semanticName ) != key.first.end() ) return true;
```

Comment:
well, `key.first.contains( semanticName )` needs C++20. But is there a way to abstract all these `if ( key.first.find( semanticName ) != key.first.end() )` repetition ?